python/IECoreDelightTest/RendererTest.py

```diff
@@ -178,13 +183,38 @@ def testMesh( self ) :
 		r.render()
 		del r
 
-		nsi = self.__parse( self.temporaryDirectory() / "test.nsi" )
+		nsi = self.__parseDict( self.temporaryDirectory() / "test.nsi" )
 
-		self.__assertInNSI( '"P.indices" "int" 8 [ 0 1 4 3 1 2 5 4 ]', nsi )
-		self.__assertInNSI( '"P" "v point" 6 [ -1 -1 0 0 -1 0 1 -1 0 -1 1 0 0 1 0 1 1 0 ]', nsi )
-		self.__assertInNSI( '"nvertices" "int" 2 [ 4 4 ]', nsi )
-		self.__assertInNSI( '"uv" "float[2]" 6 [ 0 0 0.5 0 1 0 0 1 0.5 1 1 1 ]', nsi )
-		self.__assertInNSI( '"uv.indices" "int" 8 [ 0 1 4 3 1 2 5 4 ]', nsi )
+		meshes = { k: v for k, v in nsi.items() if nsi[k]["nodeType"] == "mesh" }
+		self.assertEqual( len( meshes ), 1 )
+
+		mesh = meshes[next( iter( meshes ) ) ]
+
+		self.assertEqual( mesh["P.indices"], [ 0, 1, 4, 3, 1, 2, 5, 4 ] )
+		self.assertEqual(
+			mesh["P"],
+			[
+				imath.V3f( -1, -1, 0 ),
+				imath.V3f( 0, -1, 0 ),
+				imath.V3f( 1, -1, 0 ),
+				imath.V3f( -1, 1, 0 ),
+				imath.V3f( 0, 1, 0 ),
+				imath.V3f( 1, 1, 0 ),
+			]
+		)
+		self.assertEqual( mesh["nvertices"], [ 4, 4 ] )
+		self.assertEqual(
+			mesh["uv"],
+			[
+				imath.V2f( 0, 0 ),
+				imath.V2f( 0.5, 0 ),
+				imath.V2f( 1, 0 ),
+				imath.V2f( 0, 1 ),
+				imath.V2f( 0.5, 1 ),
+				imath.V2f( 1, 1 )
+			]
+		)
+		self.assertEqual( mesh["uv.indices"], [ 0, 1, 4, 3, 1, 2, 5, 4 ] )
```

Comment:
This is so much better than the string-based testing I started with! Thanks!

Reply:
🙌 My pleasure, it is much nicer to test against!